### ISSUE TITLE
Button.value - True / False?

### ISSUE THREAD

**Issue by martinohanlon** [lawsie/guizero#79]
Currently the value of Button returns the text which is on it.

Would it be better as a 1, 0 representing whether it was pressed or not.

**Comment by Coal0** [comment]
This is a valid question- `value` is rather inconsistent in what it returns, e.g.:

* `CheckButton` returns the button's state
* `Picture` returns the image's path
* `Slider` returns the value of the slider
* `Button`, as mentioned, returns the text value

IMO, it'd be a good idea to give these properties separate names: `Picture`: `value` → `file`. I don't know if it desirable to have a 'value' attribute common to all widgets if its implementation is inconsistent. Any thoughts?

**Comment by martinohanlon** [comment]
Having a generic value property simplifies the API and also allows for X class integration, say you wanted to display the value from a slide in a TextBox.

```
textbox.value = slider.value
```

or when you enter a file name in a text box it opens a picture.

```
picture.value = textbox.value
```

It also works well with other `zero` libraries, such as gpiozero, if Button was changed to be 1/0, you could do:

```
led.value = button.value
```

There is however nothing to say you cant duplicate properties, e.g. Picture has a `file_name` property as well as a `value` property - they return the same thing and have different abstractions for different use cases.

**Comment by Coal0** [comment]
I find that a very weird use case for properties; not entirely convinced still. It's up to you.

**Comment by martinohanlon** [comment]
I have pulled together a POC for this and it works well. It also opens up opportunities for `when_released`.

```
from tkinter import Button, PhotoImage, StringVar, DISABLED, NORMAL
from .tkmixins import ScheduleMixin, DestroyMixin, EnableMixin, FocusMixin, DisplayMixin, ReprMixin
from . import utilities as utils

class AltPushButton(ScheduleMixin, DestroyMixin, EnableMixin, FocusMixin, DisplayMixin, ReprMixin):

    def __init__(self, master, command, args=None, text="Button", icon=None, pady=10, padx=10, grid=None, align=None):

        self._text = StringVar()
        self._text.set(text)
        self._current_font = "Arial"
        self._font_size = 11
        self._value = 0
        self._command = command 

        # Description of this object (for friendly error messages)
        self.description = "[PushButton] object with text \"" + self._text.get() + "\""

        # Create a tk Button object within this object
        self.tk = Button(master.tk, textvariable=self._text)

        # If the args LIST was not blank, convert to allow args
        if args is not None:
            self._command = utils.with_args(self._command, *args)

        # Add command and padding if necessary
        self.tk.config(pady=pady, padx=padx)

        # Setup event for release
        self.tk.bind("<ButtonPress>", self._on_press)
        self.tk.bind("<ButtonRelease>", self._on_release) 

        # Try to instantiate a picture
        if icon is not None:
            try:
                self._icon = PhotoImage(file=icon)
                self.tk.config(image=self._icon)
            except AttributeError:
                utils.error_format("Image import error - image must be a gif, check correct path")


        # Pack or grid depending on parent
        try:
            utils.auto_pack(self, master, grid, align)
        except AttributeError:
            utils.error_format( self.description + "\n" +
            "Could not add to interface - check first argument is [App] or [Box]")

    # PROPERTIES
    # ----------------------------------

    # Get the value of the button (pressed, released) 
    @property
    def value(self):
        return self._value

    # Get the text on the button
    @property
    def text(self):
        return (self._text.get())

    # Set the text on the button
    @text.setter
    def text(self, value):
        self._text.set(str(value))
        self.description = "[Text] object with text \"" + str(value) + "\""

    # Get the text colour as a string
    @property
    def text_color(self):
        return (self.tk.cget("fg"))

    # Set the text colour
    @text_color.setter
    def text_color(self, color):
        self.tk.config(fg=color)

    # Get the background colour as a string
    @property
    def bg(self):
        return (self.tk.cget("bg"))

    # Set the background colour
    @bg.setter
    def bg(self, color):
        self.tk.config(bg=color)

    # Get the current font as a string
    @property
    def font(self):
        return (self._current_font)

    # Set the current font
    @font.setter
    def font(self, font):
        self._current_font = font
        self.tk.config(font=(self._current_font, self._font_size))

    # Get the current text size as an integer
    @property
    def text_size(self):
        return (self._font_size)

    # Set the font size
    @text_size.setter
    def text_size(self, size):
        self._font_size = int(size)
        self.tk.config(font=(self._current_font, self._font_size))

    # Get the current height as an integer (does not include padding)
    @property
    def height(self):
        return int(self.tk.cget("height"))

    # Set the height
    @height.setter
    def height(self, height_px):
        self.tk.config(height=int(height_px))

    # Get the current width as an integer (does not include padding)
    @property
    def width(self):
        return int(self.tk.cget("width"))

    # Set the height
    @width.setter
    def width(self, width_px):
        self.tk.config(width=int(width_px))


    # METHODS
    # -------------------------------------------

    # Internal use only
    # Called when the button is pressed
    def _on_press(self, event):
        self._value = 1
        self._command()
        
    # Called when the button is released
    def _on_release(self, event):
        self._value = 0
        
    # Change command - needs the name of a function and optional args as a list
    def change_command(self, newcommand, args=None):
        if args is not None:
            newcommand = utils.with_args(newcommand, *args)
        self.tk.config(command=newcommand)

    # Enable Button - contributed by jezdean
    """
    def enable(self):
        self.tk.config(state=NORMAL)

    # Disable Button
    def disable(self):
        self.tk.config(state=DISABLED)
    """

    def toggle(self):
        button_state = self.tk.cget("state")
        if button_state == "disabled":
            self.enable()
        elif button_state == "normal":
            self.disable()

    # Change padding
    def padding(self, padx, pady):
        self.tk.config(padx=padx, pady=pady)

    # Set the icon
    def icon(self, icon):
        try:
            img = PhotoImage(file=icon)
            self._icon = img
            self.tk.config(image=img)
        except AttributeError:
            utils.error_format("Image import error - image must be a gif, check correct path")


    # DEPRECATED
    # -------------------------------------------
    # Change text
    def set_text(self, text):
        self._text.set(str(text))
        self.description = "[Text] object with text \"" + str(text) + "\""
        utils.deprecated("PushButton set_text() is deprecated. Please use the value property instead.")

    # Toggle button state - renamed to just toggle
    def toggle_state(self):
        button_state = self.tk.cget("state")
        if button_state == "disabled":
            self.enable()
        elif button_state == "normal":
            self.disable()
        utils.deprecated("PushButton toggle_state() is deprecated - renamed to toggle()")

```

**Comment by Coal0** [comment]
May I ask you to enable syntax highlighting? It's really hard to interpret as is.

\`\`\`python
< code block >
\`\`\`

**Comment by martinohanlon** [comment]
done

**Comment by martinohanlon** [comment]
I realised something interesting.  The 'normal' button command gets fired when the button is released, not when its pressed, so I had to make a change:

```python
    # Called when the button is pressed
    def _on_press(self, event):
        self._value = 1
        
    # Called when the button is released
    def _on_release(self, event):
        self._value = 0
        self._command()
```

**Comment by martinohanlon** [comment]
I have included this change in v0.4 and updated the docs.
